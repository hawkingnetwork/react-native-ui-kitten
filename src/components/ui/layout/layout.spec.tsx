import React from 'react';
import {
  render,
  shallow,
  RenderAPI,
} from 'react-native-testing-library';
import {
  ApplicationProvider,
  ApplicationProviderProps,
} from '@kitten/theme';
import {
  Layout,
  LayoutProps,
} from './layout.component';
import {
  mapping,
  theme,
} from '../support/tests';

const Mock = (props?: LayoutProps): React.ReactElement<ApplicationProviderProps> => {
  return (
    <ApplicationProvider
      mapping={mapping}
      theme={theme}>
      <Layout {...props} />
    </ApplicationProvider>
  );
};

const renderComponent = (props?: LayoutProps): RenderAPI => {
  return render(
    <Mock {...props}/>,
  );
};

describe('@layout: matches snapshot', () => {

  it('default', () => {
    const component: RenderAPI = renderComponent();

    const { output } = shallow(component.getByType(Layout));

    expect(output).toMatchSnapshot();
  });

  it('with styles', () => {
    const component: RenderAPI = renderComponent({ style: { height: 300 } });

    const { output } = shallow(component.getByType(Layout));

    expect(output).toMatchSnapshot();
  });

});
