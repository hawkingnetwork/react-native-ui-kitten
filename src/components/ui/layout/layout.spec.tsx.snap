// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@layout: matches snapshot default 1`] = `
<Wrapper
  forwardedRef={null}
/>
`;

exports[`@layout: matches snapshot with styles 1`] = `
<Wrapper
  forwardedRef={null}
  style={
    Object {
      "height": 300,
    }
  }
/>
`;
